packages/dart_frog_auth/lib/src/dart_frog_auth.dart

```diff
@@ -47,27 +47,42 @@ Future<bool> _defaultApplies(RequestContext context) async => true;
 /// [RequestContext]. If the function returns false, the middleware will not
 /// apply to the route and the call will have authentication validation.
 Middleware basicAuthentication<T extends Object>({
-  required Future<T?> Function(String, String) userFromCredentials,
+  Future<T?> Function(String, String)? userFromCredentials,
+  Future<T?> Function(String, String, RequestContext)? userFromContext,
```

Comment:
I agree that we need to work on the naming. I feel that after reading `userFromCredentials`, I get the feeling that `userFromContext` does not provide any credentials, instead these credentials are in the context.

In addition, to keep consistency with Flutter (for example, ListView.builder's itemBuilder) I would always make the context the first parameter.

We could, once again, take inspiration from Flutter and rename the `userFromContext` to `userBuilder`.  The builder receives the `context` and the credentials as parameters. Personally, I think in the long term we can deprecate the old API.

Reply:
I understand what about: `userReader`?